presto-main/src/main/java/com/facebook/presto/sql/planner/SqlPlannerContext.java

```diff
@@ -49,4 +63,58 @@ public void incrementLeafNodes(Session session)
             }
         }
     }
+
+    public class NestedCteStack
```

Comment:
NestedCteStack is responsible for managing the scope of the CTE
 cteStack keeps track of the current CTEs that are accessible outside this scope.
 The rawCtePathMap holds the localCtes along with their respective scopes, keeping them updated.